misc/windows-deploy/hostsetup.ps1

```diff
@@ -11,21 +11,12 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-$networkAdapters = Get-CimInstance -ClassName Win32_NetworkAdapter -Filter "AdapterTypeId='0' AND NetEnabled='True' AND NOT Name LIKE '%Hyper-V Virtual%' AND NOT Name LIKE '%Loopback Adapter%' AND NOT Name LIKE '%TAP-Windows Adapter%'" | Sort-Object -Property "Index"
-if(-not $networkAdapters -or $networkAdapters.Length -eq 0)
-{
-    throw New-Object System.Exception("Failed to find the primary network interface")
-}
-$ifIndex = $networkAdapters[0].InterfaceIndex
-$networkAdapterConfig = Get-CimInstance -ClassName Win32_NetworkAdapterConfiguration -Filter "InterfaceIndex='$ifIndex'" | select IPConnectionMetric, DefaultIPGateway
-$defaultGateway = $networkAdapterConfig.DefaultIPGateway[$networkAdapterConfig.DefaultIPGateway.Length - 1]
+$ErrorActionPreference = 'Stop'
 
-$containerGateway = (Get-NetNat).InternalIPInterfaceAddressPrefix | %{ $_ -replace '/.*$', '' }
-
-netsh interface portproxy add v4tov4 listenaddress=$containerGateway listenport=51679 connectaddress=127.0.0.1 connectport=51679 protocol=tcp
-
-$existingRoute=route print 169.254.170.2 -4 | Sort-Object -Property "IPv4 Route Table"
-if ($existingRoute -Match "169.254.170.2") {
-   route -p delete 169.254.170.2/32
-}
-New-NetRoute -DestinationPrefix 169.254.170.2/32 -InterfaceIndex $ifIndex -NextHop $defaultGateway
+# Enable NAT for container-to-host communication
+Add-VMNetworkAdapter -VMNetworkAdapterName "APIPA" -SwitchName nat -ManagementOS
+$ifIndex = (Get-NetAdapter -Name "*APIPA*" | Sort-Object | Select ifIndex).ifindex
+New-NetIPAddress -InterfaceIndex $ifIndex -IPAddress 169.254.170.2 -PrefixLength 32
+New-NetRoute -DestinationPrefix 172.20.128.0/20 -ifIndex $ifindex
```

Comment:
I do not think we should hard code the 172.20.128.0/20 here. The value corresponds to the Subnet CIDR under the IPAM  section for the default nat network. Essentially, obtained from the output of docker network inspect nat (default network). I am thinking we should discover the value than hardcode it here.